autoscale/src/main/resources/definitions/aws-vm.json

```diff
@@ -0,0 +1,174 @@
+{
```

Comment:
Can this and azure-vm.json use the same values that are already available in Cloudbreak?
Maintaining 2 copies of the same / similar files will lead to confusion sometime in the future when instances are changed.

Reply:
CB does not have  cpu, memory and similar metadata for azure, and there is no direct api to fetch it from CB. The lists are partially present for aws but it is deep inside aws connector. Only supported model names are available at top level api which is validated against periscope definition. Also going forward I think periscope should reduce it's dependency for metadata on CB to avoid unnecessary CB hits.

Reply:
https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instance-types.html
or https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/ec2/AbstractAmazonEC2.html#describeInstanceTypes-com.amazonaws.services.ec2.model.DescribeInstanceTypesRequest-
could probably be used for AWS.

http://azure.github.io/ref-docs/java/com/microsoft/azure/management/compute/VirtualMachines.html#sizes--
for Azure.

These obviously won't be used at Runtime - but could potentially be part of the build to generate this list dynamically. (Future fix).

Curious: How did you end up generating this list for aws and azure?

Also, would this end up breaking YARN. That's something that will need to be tracked down sooner rather than later, given that a lot of the testing is going to happen on YCloud.

Reply:
I used a java client with my cloud credentials to generate the list.